pkg/kubelet/dockertools/manager.go

```diff
@@ -1813,13 +1813,20 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 	if containerChanges.StartInfraContainer && (len(containerChanges.ContainersToStart) > 0) {
 		glog.V(4).Infof("Creating pod infra container for %q", podFullName)
 		podInfraContainerID, err = dm.createPodInfraContainer(pod)
+		if err != nil {
+			glog.Errorf("Failed to create pod infra container: %v; Skipping pod %q", err, podFullName)
+			return err
+		}
 
 		// Call the networking plugin
-		if err == nil {
-			err = dm.networkPlugin.SetUpPod(pod.Namespace, pod.Name, podInfraContainerID)
-		}
+		err = dm.networkPlugin.SetUpPod(pod.Namespace, pod.Name, podInfraContainerID)
 		if err != nil {
 			glog.Errorf("Failed to create pod infra container: %v; Skipping pod %q", err, podFullName)
+			// Delete infra container
+			deleteError := dm.KillContainerInPod(types.UID(podInfraContainerID), nil, pod)
```

Comment:
nit: keep the deleteError in the `if` block if you are not going to use 
`if err := dm.KillContainerInPod(types.UID(podInfraContainerID), nil, pod); err != nil {`